src/compiler/types.ts

```diff
@@ -6523,6 +6523,7 @@ namespace ts {
         ES2020 = 7,
         ES2021 = 8,
         ES2022 = 9,
+        ES2023 = 10,

```

Comment:
@DanielRosenwasser: Do we normally add a new script target before that edition is ratified, or do we put everything in `esnext` until there's an official spec?

Reply:
I think it's okay to put it into 2023 since we're expecting ratification before the final release of TS5.0.